packages/cli/src/api/split.ts

```diff
@@ -33,15 +33,22 @@ export function split(payload: z.infer<typeof splitSchema>) {
     const targetDir = path.dirname(payload.entrypointPath);
     const annotationDirectory = path.join(outputDir, index.toString());
 
-    files.forEach((file) => {
-      const relativeFileNamePath = path.relative(targetDir, file.path);
-      const destinationPath = path.join(
-        annotationDirectory,
-        relativeFileNamePath,
-      );
-      fs.mkdirSync(path.dirname(destinationPath), { recursive: true });
-      fs.writeFileSync(destinationPath, file.sourceCode, "utf8");
-    });
+    files
+      .then((files) => {
+        files.forEach((file) => {
+          const relativeFileNamePath = path.relative(targetDir, file.path);
+          const destinationPath = path.join(
+            annotationDirectory,
+            relativeFileNamePath,
+          );
+          fs.mkdirSync(path.dirname(destinationPath), { recursive: true });
+          fs.writeFileSync(destinationPath, file.sourceCode, "utf8");
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        throw error;
+      });
```

Comment:
The response is sent before the worker or done with the jobs.
Should wait until the worker are done before sending the response

Reply:
Updated the code to wait for the response 🟢